README.md

```diff
@@ -58,6 +61,7 @@ This file contains a number of Scala interview questions that can be used when v
   * How do you ensure that the compiler optimizes the tail recursive function?
 * What is function currying?
 * What are implicit parameters?
+* What does tail recursion imply? How does it differentiate from common recursion?
```

Comment:
There are already a some questions about tail recursion.
- I think that we can merge "What does tail recursion imply?" and "What is tail recursion?" in a single question.
- I really like "How does it differentiate from common recursion?". Can you move this question with the rest of them?